rest-service/manager_rest/test/security/test_userstore_reload.py

```
@@ -0,0 +1,104 @@
+
+
+@attr(client_min_version=2, client_max_version=base_test.LATEST_API_VERSION)
+class TestUserstoreReloadFile(SecurityTestBase):
+
+    def setUp(self):
+        self.modify_userstore_file(self.get_users(), self.get_groups())
+        super(TestUserstoreReloadFile, self).setUp()
+
+    def test_reload_userstore(self):
+        self.modify_userstore_file(self.get_users(),
+                                   self.get_groups(),
+                                   wait_for_loading=3)
+        client = self.create_client(headers=SecurityTestBase.
+                                    create_auth_header(
+                                        username='alice',
+                                        password='alice_password'))
+        # alice should be able to do everything
+        client.deployments.list()
+
+        # modify the userstore file. Change alice's password
+        new_file_users = [
+            {
+                'user': 'alice',
+                'password': 'non_existing_password'
+            }
+        ]
+        self.modify_userstore_file(new_file_users,
+                                   [],
+                                   wait_for_loading=3)
+
+        # assert alice's old credentials are no longer valid
+        self.assertRaises(UserUnauthorizedError, client.deployments.list)
+
+    def test_reload_invalid_userstore(self):
+        self.modify_userstore_file(self.get_users(),
+                                   self.get_groups(),
+                                   wait_for_loading=3)
+        client = self.create_client(headers=SecurityTestBase.
+                                    create_auth_header(
+                                        username='alice',
+                                        password='alice_password'))
+        # alice should be able to do everything
+        client.deployments.list()
+
+        # Corrupt userstore file
+        self.corrupt_userstore_file(wait_for_loading=3)
```

an exception should be throws, let's verify it's not a silent failure